File: app/src/main/java/com/kotoba/takarabako/data/model/Quote.kt
package com.kotoba.takarabako.data.model

data class QuoteSegment(
    val kanji: String,
    val reading: String
)

data class Quote(
    val id: String,
    val cat: String,
    val kanji: String,
    val segments: List<QuoteSegment>,
    val korean: String,
    val author: String
)

File: KotobaNoTakarabako-iOS/Models/Quote.swift
import Foundation

struct Segment: Codable, Hashable {
    let text: String
    let reading: String

    var hasReading: Bool { !reading.isEmpty }
}

struct Quote: Codable, Identifiable, Hashable {
    let id: String
    let cat: String
    let kanji: String
    let segments: [Segment]
    let korean: String
    let author: String
}
